nifi-assembly/pom.xml

```diff
@@ -451,356 +218,597 @@ language governing permissions and limitations under the License. -->
-            <type>nar</type>
-        </dependency>
-        <dependency>
-            <groupId>org.apache.nifi</groupId>
-            <artifactId>nifi-livy-controller-service-api-nar</artifactId>
-            <version>1.13.0-SNAPSHOT</version>
-            <type>nar</type>
-        </dependency>
-        <dependency>
-            <groupId>org.apache.nifi</groupId>
-            <artifactId>nifi-livy-nar</artifactId>
-            <version>1.13.0-SNAPSHOT</version>
-            <type>nar</type>
-        </dependency>
-        <dependency>
-            <groupId>org.apache.nifi</groupId>
-            <artifactId>nifi-kerberos-credentials-service-nar</artifactId>
-            <version>1.13.0-SNAPSHOT</version>
-            <type>nar</type>
-        </dependency>
-        <dependency>
-            <groupId>org.apache.nifi</groupId>
-            <artifactId>nifi-proxy-configuration-nar</artifactId>
-            <version>1.13.0-SNAPSHOT</version>
-            <type>nar</type>
-        </dependency>
-        <dependency>
-            <groupId>org.apache.nifi</groupId>
-            <artifactId>nifi-jolt-record-nar</artifactId>
-            <version>1.13.0-SNAPSHOT</version>
-            <type>nar</type>
-        </dependency>
-        <dependency>
-            <groupId>org.apache.nifi</groupId>
-            <artifactId>nifi-record-sink-service-nar</artifactId>
-            <version>1.13.0-SNAPSHOT</version>
-            <type>nar</type>
-        </dependency>
-        <!-- dependencies for jaxb/activation/annotation for running NiFi on Java 11 -->
-        <!-- TODO: remove these once minimum Java version is 11 -->
-        <dependency>
-            <groupId>javax.xml.bind</groupId>
-            <artifactId>jaxb-api</artifactId>
-            <version>2.3.0</version>
-        </dependency>
-        <dependency>
-            <groupId>com.sun.xml.bind</groupId>
-            <artifactId>jaxb-impl</artifactId>
-            <version>2.3.0</version>
-        </dependency>
-        <dependency>
-            <groupId>com.sun.xml.bind</groupId>
-            <artifactId>jaxb-core</artifactId>
-            <version>2.3.0</version>
-        </dependency>
-        <dependency>
-            <groupId>javax.annotation</groupId>
-            <artifactId>javax.annotation-api</artifactId>
-            <version>1.3.2</version>
-        </dependency>
-        <dependency>
-            <groupId>javax.activation</groupId>
-            <artifactId>javax.activation-api</artifactId>
-            <version>1.2.0</version>
-        </dependency>
-        <!-- AspectJ library needed by the Java Agent used for native library loading (see bootstrap.conf) -->
-        <dependency>
-            <groupId>org.aspectj</groupId>
-            <artifactId>aspectjweaver</artifactId>
+            <groupId>org.aspectj</groupId>
+            <artifactId>aspectjweaver</artifactId>
         </dependency>
     </dependencies>
     <profiles>
+        <profile>
+            <id>most-nars</id>
```

Comment:
Two thoughts here:

1. Might make more sense to people if we called it something like `distribution-nars` to helpfully indicate these are the standard distribution NARs.
2. What do you think about breaking this profile up into half a dozen or more autoactivated profiles that make it easier for people to exclude things?

Reply:
I'd love (2) but agreeing on a breakdown might be hard. Nonetheless, I'll offer up what I'd find helpful (ignoring dependencies):
- Hadoop and friends (excluding Parquet)
- Cloud X
- Cloud Y
- Cloud Z
- Messaging
- Databases
- Scripting
- Misc